source/hash_table.c

```diff
@@ -653,38 +670,61 @@ static inline void s_get_next_element(struct aws_hash_iter *iter, size_t start_s
         if (entry->hash_code) {
             iter->element = entry->element;
             iter->slot = i;
+            iter->status = AWS_HASH_ITER_STATUS_READY_FOR_USE;
             return;
         }
     }
     iter->element.key = NULL;
     iter->element.value = NULL;
     iter->slot = iter->limit;
+    iter->status = AWS_HASH_ITER_STATUS_DONE;
+    AWS_POSTCONDITION(aws_hash_iter_is_valid(iter));
 }
 
 struct aws_hash_iter aws_hash_iter_begin(const struct aws_hash_table *map) {
+    AWS_PRECONDITION(aws_hash_table_is_valid(map));
     struct hash_table_state *state = map->p_impl;
     struct aws_hash_iter iter;
     iter.map = map;
     iter.limit = state->size;
     s_get_next_element(&iter, 0);
+    AWS_POSTCONDITION(iter.status == AWS_HASH_ITER_STATUS_DONE || iter.status == AWS_HASH_ITER_STATUS_READY_FOR_USE);
+    AWS_POSTCONDITION(aws_hash_iter_is_valid(&iter));
     return iter;
 }
 
 bool aws_hash_iter_done(const struct aws_hash_iter *iter) {
+    AWS_PRECONDITION(iter);
```

Comment:
This should be checked in `aws_hash_iter_is_valid(iter)`

Reply:
Done